vnext/ReactWindowsCore/WinRTWebSocketResource.cpp

```diff
@@ -0,0 +1,363 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// clang-format off
+#include "WinRTWebSocketResource.h"
+
+#include <winrt/Windows.Security.Cryptography.h>
+#include <Unicode.h>
+#include <Utilities.h>
+
+// Standard Library
+#include <future>
+
+using Microsoft::Common::Unicode::Utf8ToUtf16;
+using Microsoft::Common::Unicode::Utf16ToUtf8;
+using Microsoft::Common::Utilities::CheckedReinterpretCast;
+
+using std::function;
+using std::size_t;
+using std::string;
+using std::vector;
+
+using winrt::fire_and_forget;
+using winrt::hresult;
+using winrt::hresult_error;
+using winrt::resume_background;
+using winrt::Windows::Foundation::Uri;
+using winrt::Windows::Networking::Sockets::IWebSocket;
+using winrt::Windows::Networking::Sockets::MessageWebSocket;
+using winrt::Windows::Networking::Sockets::MessageWebSocketMessageReceivedEventArgs;
+using winrt::Windows::Networking::Sockets::SocketMessageType;
+using winrt::Windows::Networking::Sockets::WebSocketClosedEventArgs;
+using winrt::Windows::Security::Cryptography::CryptographicBuffer;
+using winrt::Windows::Storage::Streams::DataReader;
+using winrt::Windows::Storage::Streams::DataWriter;
+using winrt::Windows::Storage::Streams::UnicodeEncoding;
+
+namespace Microsoft::React
+{
+WinRTWebSocketResource::WinRTWebSocketResource(Uri&& uri)
+  : m_uri{ std::move(uri) }
+  , m_socket{}
+  , m_writer{ m_socket.OutputStream() }
```

Comment:
This is dangerous because this initialization order is based on the member variables order declared in class, not on the order it's specified in constructor initialization. It's unfortunate that C++ has no warning for this. Please move this initialization to .h for clarity.